### ISSUE TITLE
Compile Error when using i2c_smbus_write_i2c_block_data

### ISSUE THREAD

**Issue by tomdean1** [raspberrypi/tools#17]
Please put the latest i2c-dev.h in the tool chain.

To fix my compile problem, I copied the latest i2c-dev. h into the tool chain.

cp /usr/include/linux/i2c-dev.h ./arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64/arm-linux-gnueabihf/libc/usr/include/linux/i2c-dev.h 

Now, compile works and the resulting image runs properly on the RPi.

Before replacing i2c-dev.h, 

> make PROGXX=lis3lv02
/home/tomdean/RaspberryPi/tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64/bin/arm-linux-gnueabihf-g++ -I../include -I. -Os -mcpu=arm1176jzf-s -mfpu=vfp -mfloat-abi=hard -marm    -c -o lis3lv02.o lis3lv02.cc
lis3lv02.cc: In constructor 'LIS3LV02::LIS3LV02()':
lis3lv02.cc:33:59: error: 'i2c_smbus_write_i2c_block_data' was not declared in this scope
   res = i2c_smbus_write_i2c_block_data(device, reg, 2, buf);
                                                           ^
lis3lv02.cc: In member function 'void LIS3LV02::read_accel()':
lis3lv02.cc:71:57: error: 'i2c_smbus_read_i2c_block_data' was not declared in this scope
  res = i2c_smbus_read_i2c_block_data(device, reg, 6, buf);
                                                         ^
make: *** [lis3lv02.o] Error 1

**Comment by Ruffio** [comment]
@popcorntime is this an easy and correct fix?

**Comment by tomdean1** [comment]
On 06/27/15 00:00, Rasmus Christiansen wrote:
> @popcorntime <https://github.com/popcorntime> is this an easy and
> correct fix?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/raspberrypi/tools/issues/17#issuecomment-115975938>.
>


Worked for me.

Tom Dean

**Comment by popcornmix** [comment]
Not sure it's correct. The toochain contains a set a executables, libraries and includes that need to be consistent. While replacing a single include file might work for reporter's specific case, I suspect other use cases will require other files to be changed, or that change reverted.

I suspect ensuring /usr/include/linux is on the include path before arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64/arm-linux-gnueabihf/libc/usr/include/linux is the correct solution.

Also arm-linux-gnueabihf-gcc is the latest toolchain in this repo, so that should be used instead.

**Comment by tomdean1** [comment]
This problem still exists.  There does not appear to be a work-around other than that I proposed in the original comment.

I am cross-compiling.  I cloned into the latest tools. 

make PROGXX=lis3lv02
/home/tomdean/RaspberryPi/tools/arm-bcm2708/arm-bcm2708-linux-gnueabi/bin/arm-bcm2708-linux-gnueabi-g++ -I../include -I. -Os -mcpu=arm1176jzf-s -mfpu=vfp -mfloat-abi=hard -marm    -c -o lis3lv02.o lis3lv02.cc
lis3lv02.cc: In constructor 'LIS3LV02::LIS3LV02()':
lis3lv02.cc:33:59: error: 'i2c_smbus_write_i2c_block_data' was not declared in this scope
lis3lv02.cc: In member function 'void LIS3LV02::read_accel()':
lis3lv02.cc:71:57: error: 'i2c_smbus_read_i2c_block_data' was not declared in this scope

Placing /usr/include/linux before arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64/arm-linux-gnueabihf/libc/usr/include/linux results in many errors.

Is there a proper solution?
